### PR TITLE
feat(config): support configure input length

### PR DESCRIPTION
**Pull request by xxs-wallace**
close https://github.com/TabbyML/tabby/issues/2638

### CODE REVIEW

File: crates/llama-cpp-server/src/supervisor.rs [hunk 1]
```diff
@@ -25,6 +25,7 @@ impl LlamaCppSupervisor {
         parallelism: u8,
         chat_template: Option<String>,
         enable_fast_attention: bool,
+        llama_context_size: usize,
```
Comment (wsxiaoys) [inline crates/llama-cpp-server/src/supervisor.rs:28]:
```suggestion
        context_size: usize,
```

File: crates/tabby-common/src/config.rs [hunk 2]
```diff
@@ -186,6 +187,8 @@ pub struct ModelConfigGroup {
     pub chat: Option<ModelConfig>,
     #[serde(default = "default_embedding_config")]
     pub embedding: ModelConfig,
+
+    pub completion_max_input_length: usize,
```
Comment (wsxiaoys) [inline crates/tabby-common/src/config.rs:191]: need `serde(default=...)`?

File: crates/tabby/src/main.rs [hunk 3]
```diff
@@ -136,5 +136,6 @@ fn to_local_config(model: &str, parallelism: u8, device: &Device) -> ModelConfig
         parallelism,
         num_gpu_layers,
         enable_fast_attention: None,
+        context_size: 4096,
```
Comment (wsxiaoys) [inline crates/tabby/src/main.rs:139]: is there a way to reuse default definition in `config.rs`?

File: crates/tabby/src/services/completion.rs [hunk 4]
```diff
@@ -230,6 +230,7 @@ pub struct CompletionService {
     engine: Arc<CodeGeneration>,
     logger: Arc<dyn EventLogger>,
     prompt_builder: completion_prompt::PromptBuilder,
+    complete_max_input_length: usize,
```
Comment (wsxiaoys) [inline crates/tabby/src/services/completion.rs:233]:
```suggestion
    max_input_length: usize,
```

**Review by wsxiaoys — CHANGES_REQUESTED** [review]

File: crates/tabby/src/main.rs (hunk 3 above)
Comment (xxs-wallace) [reply crates/tabby/src/main.rs]: yes, but we must use it as a public function

File: crates/llama-cpp-server/src/supervisor.rs (hunk 1 above)
Comment (xxs-wallace) [reply crates/llama-cpp-server/src/supervisor.rs]: done

**Pull request merged**